src/main/java/org/opentripplanner/graph_builder/module/osm/naming/SidewalkNamer.java

```diff
@@ -0,0 +1,290 @@
+ *    the buffer and apply its name to the sidewalk.
+ * <p>
+ * This works very well for OSM data where the sidewalk runs a parallel to the street and at each
+ * intersection the sidewalk is also split. It doesn't work well for sidewalks that go around
+ * the corner, like https://www.openstreetmap.org/way/1059101564. These cases are, however, detected
+ * by the above algorithm and the sidewalk name remains the same.
+ */
+public class SidewalkNamer implements EdgeNamer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SidewalkNamer.class);
+  private static final double MIN_PERCENT_IN_BUFFER = .85;
+  private static final int BUFFER_METERS = 25;
+
+  private HashGridSpatialIndex<EdgeOnLevel> streetEdges = new HashGridSpatialIndex<>();
+  private Collection<EdgeOnLevel> unnamedSidewalks = new ArrayList<>();
+  private PreciseBuffer preciseBuffer;
+
+  @Override
+  public I18NString name(OSMWithTags way) {
+    return way.getAssumedName();
+  }
+
+  @Override
+  public void recordEdges(OSMWithTags way, StreetEdgePair pair) {
+    // this way is a sidewalk and hasn't been named yet (and is not explicitly unnamed)
+    if (way.isSidewalk() && way.hasNoName() && !way.isExplicitlyUnnamed()) {
+      pair
+        .asIterable()
+        .forEach(edge -> unnamedSidewalks.add(new EdgeOnLevel(edge, way.getLevels())));
+    }
+    // the way is _not_ a sidewalk and does have a name
+    else if (way.isNamed() && !way.isLink()) {
+      // We generate two edges for each osm way: one there and one back. This spatial index only
+      // needs to contain one item for each road segment with a unique geometry and name, so we
+      // add only one of the two edges.
+      var edge = pair.pickAny();
+      streetEdges.insert(
+        edge.getGeometry().getEnvelopeInternal(),
+        new EdgeOnLevel(edge, way.getLevels())
+      );
+    }
+  }
+
+  @Override
+  public void postprocess() {
+    ProgressTracker progress = ProgressTracker.track(
+      "Assigning names to sidewalks",
+      500,
+      unnamedSidewalks.size()
+    );
+
+    this.preciseBuffer = new PreciseBuffer(computeEnvelopeCenter(), BUFFER_METERS);
+
+    final AtomicInteger namesApplied = new AtomicInteger(0);
+    unnamedSidewalks
+      .parallelStream()
+      .forEach(sidewalkOnLevel -> {
+        assignNameToSidewalk(sidewalkOnLevel, namesApplied);
+
+        // Keep lambda! A method-ref would cause incorrect class and line number to be logged
+        // noinspection Convert2MethodRef
+        progress.step(m -> LOG.info(m));
+      });
+
+    LOG.info(
+      "Assigned names to {} of {} of sidewalks ({}%)",
+      namesApplied.get(),
+      unnamedSidewalks.size(),
+      DoubleUtils.roundTo2Decimals((double) namesApplied.get() / unnamedSidewalks.size() * 100)
+    );
+
+    LOG.info(progress.completeMessage());
+
+    // set the indices to null so they can be garbage-collected
```

Comment:
Capitalize first letter of sentence.

Reply:
https://github.com/opentripplanner/OpenTripPlanner/pull/5774/commits/60c4ee925f51689bacd596f517ea709984b36615